contrib/flesctl/init_run.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python3
+
+import os
+import subprocess
+import datetime
+import argparse
+import flescfg
+
+def main():
+    parser = argparse.ArgumentParser(description='Generate Flesnet and SPM configurations.')
+    parser.add_argument('config', help='Path to the configuration file')
+    parser.add_argument('run_id', help='Run ID')
+    args = parser.parse_args()
+
+    config_file = args.config
+    run_id = args.run_id
+
+    # global parameters, may be overwritten by environment
+    FLESNETDIR = os.getenv('FLESNETDIR', '/usr/bin/')
+    FLESCTLDIR = os.getenv('FLESCTLDIR', '/opt/fles/bin/')
+    SPMDIR = os.getenv('SPMDIR', '/opt/spm/')
+    RUNDIR = os.getenv('RUNDIR', '')
+    LOGDIR = os.getenv('LOGDIR', 'log/')
+    FLESNET_CFG = os.getenv('FLESNET_CFG', os.path.join(RUNDIR, 'flesnet.cfg'))
+    SPM_CFG = os.getenv('SPM_CFG', os.path.join(RUNDIR, 'readout.spm'))
+
+    # default configuration
+    TSBUF_DATA_SIZE_EXP = 27
+    TSBUF_DESC_SIZE_EXP = 19
+
+    # load the config file
+    config = flescfg.load(config_file)
+    if config is None:
+        print("Error loading configuration file.")
+        exit(1)
+    
+    INPUT = config['use_entry_nodes']
+    OUTPUT = config['use_build_nodes']
+    TRANSPORT = config['common']['transport']
+    TIMESLICE_SIZE = config['common']['timeslice_size']
+    OVERLAP_SIZE = config['common']['timeslice_overlap']
+    TSCLIENT_PARAM = config['common']['tsclient_param']
+    EXTRA_CMD = config['common']['extra_cmd']
+
+    # read node list into python lists
+    EN_LIST = subprocess.check_output(["scontrol", "show", "hostname", INPUT]).decode().split()
+    PN_LIST = subprocess.check_output(["scontrol", "show", "hostname", OUTPUT]).decode().split()
+
+    print("Generating configs for", EN_LIST, PN_LIST)
+    print("Writing output to", FLESNET_CFG, "and", SPM_CFG)
+
+    # entry nodes
+    EN_CFG = []
+    EN_INDEX = []
+    EN_SPM = []
+    CRI_SPM = []
+    input = 0
+
+    SHM_PREFIX = 'cri_'
+    for node in EN_LIST:
+        nodeinfo = config["entry_nodes"][node]
+        for card in nodeinfo['cards']:
+            cardinfo = nodeinfo['cards'][card]
+            shm_channel = 0
+            for channel in range(8):
+                if channel in cardinfo["channels"]:
+                    EN_CFG.append(f"input = shm://{nodeinfo['address']}/{SHM_PREFIX}{card}/{shm_channel}?overlap={OVERLAP_SIZE}")
+                    EN_INDEX.append(f"-i{input}")
+                    shm_channel += 1
+                    input += 1
+            EN_CFG.append("")
+            CRI_SPM.append(f"cri_{node}_{cardinfo['pci_address'][:2]}")
+        EN_SPM.append(f"{node}: bash -c \"LOGDIR={LOGDIR}{node}_ {FLESCTLDIR}/fles_input {config_file} {node}\"")
```

Comment:
```suggestion
        EN_SPM.append(f"{node}: bash -c \"LOGDIR={LOGDIR}{node}_ {FLESCTLDIR}/fles_input.py {config_file} {node}\"")
```
Hi Jan

If I understood your scripts and what I saw in the final diff of the compare tool correctly, the Python version of the input script should be used if the input is yaml, is that right?